pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotLeadControllerRestletResource.java

```diff
@@ -0,0 +1,197 @@
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.controller.api.resources;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import javax.inject.Inject;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import org.apache.helix.HelixManager;
+import org.apache.helix.model.ExternalView;
+import org.apache.helix.model.MasterSlaveSMD;
+import org.apache.pinot.common.config.TableNameBuilder;
+import org.apache.pinot.common.utils.CommonConstants.Helix;
+import org.apache.pinot.common.utils.helix.LeadControllerUtils;
+import org.apache.pinot.controller.helix.core.PinotHelixResourceManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+@Api(tags = Constants.LEAD_CONTROLLER_TAG)
+@Path("/leader")
+public class PinotLeadControllerRestletResource {
+  public static Logger LOGGER = LoggerFactory.getLogger(PinotLeadControllerRestletResource.class);
+
+  @Inject
+  private PinotHelixResourceManager _pinotHelixResourceManager;
+
+  @GET
+  @Produces(MediaType.APPLICATION_JSON)
+  @Path("/tables")
+  @ApiOperation(value = "Gets leaders for all tables", notes = "Gets leaders for all tables")
+  public LeadControllerResponse getLeadersForAllTables() {
+    Map<String, LeadControllerEntry> leadControllerEntryMap = new LinkedHashMap<>();
+    HelixManager helixManager = _pinotHelixResourceManager.getHelixZkManager();
+    boolean isLeadControllerResourceEnabled = LeadControllerUtils.isLeadControllerResourceEnabled(helixManager);
+
+    String helixLeader = null;
+    // Returns helix leader if lead controller resource is disabled.
+    if (!isLeadControllerResourceEnabled) {
+      helixLeader = LeadControllerUtils.getHelixClusterLeader(helixManager);
+    }
+
+    ExternalView leadControllerResourceExternalView = getLeadControllerResourceExternalView(helixManager);
```

Comment:
Re-thinking, it is better to return an empty map when resource is disabled.  We can decouple helix leadership from this one